### PR TITLE
Saurabhj store implementation

### PR DESCRIPTION
**Pull request by saurabhjinturkar**

### CODE REVIEW

File: resources/storage.proto [hunk 1]
```diff
@@ -11,7 +34,42 @@ enum Action {
...
+	required int32 seq_size = 19;
+	optional int32 size = 20;
+	optional int64 time = 21;
+}
+
+
+/*
+ * ============Query============
+ * 
+ * Query is command for the cluster. It should contain
```
Comment (codepenman) [inline resources/storage.proto:67]: Can't query be used for internal communication ?

File: resources/storage.proto [hunk 2]
```diff
@@ -11,7 +34,42 @@ enum Action {
...
+ * size attribute is for storing file size in bytes. (optional) time will
+ * store creation time. (optional)
+ *
+ * Metadata chunk should be created by client program. To create it, client
+ * should first identify file size and how many chunks will be created from
+ * the file.
+ *
+ * Metadata should be stored as sequenceNo 0.
+ */
+message Metadata {
```
Comment (codepenman) [inline resources/storage.proto:57]: Isn't this entirely a separate message which will be requested first and then data request starts? Or is it necessary to always have it in seq 0 ?

File: resources/storage.proto (hunk 1 above)
Comment (saurabhjinturkar) [inline resources/storage.proto:67]: Query can be used for internal communication. Workmessage and worktask should carry this query. But still it is a command (action) for the cluster.

File: resources/storage.proto (hunk 2 above)
Comment (saurabhjinturkar) [inline resources/storage.proto:57]: Metadata is 0 index chunk.

**Pull request merged**